wrappers/kn/src/nativeMain/kotlin/zxingcpp/BarcodeReader.kt

```diff
@@ -0,0 +1,19 @@
+package zxingcpp
+
+import kotlinx.cinterop.ExperimentalForeignApi
+import zxingcpp.cinterop.zxing_ReadBarcode
+import zxingcpp.cinterop.zxing_ReadBarcodes
+
+@OptIn(ExperimentalForeignApi::class, ExperimentalStdlibApi::class)
+class BarcodeReader {
+	fun readBarcode(imageView: ImageView, options: ReaderOptions? = null): Barcode? =
+		imageView.cValueWrapped.use {
+			zxing_ReadBarcode(it.cValue, options?.cValue)?.toKObject()
+		}
+
+	fun readBarcodes(imageView: ImageView, options: ReaderOptions? = null): List<Barcode> =
+		imageView.cValueWrapped.use {
+			zxing_ReadBarcodes(it.cValue, options?.cValue)?.toKObject()
+				?: error("zxing_ReadBarcodes returned null, which is an unexpected behaviour")
```

Comment:
That is not unexpected. If you pass in invalid data inside the ImageView parameter, then this function returns `NULL` and you need to call the `zxing_LastErrorMsg()` function to get info about what went wrong.

Reply:
fixed in https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/146966c81e97777ba5e64ba9da261dd93f72024d